lib/Dialect/Verif/VerifOps.cpp

```diff
@@ -159,6 +159,37 @@ LogicalResult BoundedModelCheckingOp::verifyRegions() {
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// SimulationOp
+//===----------------------------------------------------------------------===//
+
+LogicalResult SimulationOp::verifyRegions() {
+  if (getBody()->getNumArguments() != 2)
+    return emitOpError() << "must have two block arguments";
+  auto clockType = seq::ClockType::get(getContext());
+  auto i1Type = IntegerType::get(getContext(), 1);
+  if (getBody()->getArgumentTypes() != std::array<Type, 2>{clockType, i1Type})
```

Comment:
Nit: Maybe cheaper to do `isa<ClockType>(type)` and `type.isSignlessInteger(1)` checks rather than building new types?

Reply:
Nice idea, done